lib/rules.ml

```diff
@@ -423,6 +453,7 @@ type 'a move =
   > >) move
 
   | Gate: (<
+    count: Nat.o -> 'count;
```

Comment:
I think that if here (and in `Win`) you expect `count` to be `Nat.z` instead, you can prevent winning after teleportation even with `-rectypes`, as the infinite type `(Nat.o -> 'a) as 'a` won't be compatible with it.